### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: library/sequencer/src/lib/ui/sequence.rb [hunk 1]
```diff
@@ -6,6 +6,73 @@ module UI
...
+  #       "step1" => {
+  #         next: "step2",
+  #       },
+  #       "step2" => {
+  #         next: :finish
+  #       }
+  #     }
+  #
+  #     def step1
+  #       :next
```
Comment (mvidner) [inline library/sequencer/src/lib/ui/sequence.rb]: What about adding `# ... show UI of this step ...` above the return value?

File: library/sequencer/src/lib/ui/sequence.rb [hunk 2]
```diff
@@ -6,6 +6,73 @@ module UI
...
+  #       super(sequence: SEQUENCE_HASH)
+  #     end
+  #   end
+  #
+  # @example complex sequence with branching and auto steps that are skipped when going back
+  #   class ComplexSequence < UI::Sequence
+  #     SEQUENCE_HASH = {
+  #       START   => "step1",
+  #       "step1" => {
+  #         next: "auto_step",
```
Comment (mvidner) [inline library/sequencer/src/lib/ui/sequence.rb]: For readability I would collapse the next-only entries to a single line:
```rb
 "step1" => { next: "auto_step" },
```

File: library/sequencer/src/lib/ui/sequence.rb [hunk 3]
```diff
@@ -24,13 +91,38 @@ def self.run(aliases, sequence)
     # - auto :abort (see {#abortable})
     # - *aliases* are assumed to be method names if unspecified
     #   (see {#from_methods})
+    #
+    # @example sequence with branching with child having methods step1 and step2, where step 1 can finish sequence or run step 2
+    #   run({
```
Comment (mvidner) [inline library/sequencer/src/lib/ui/sequence.rb]:
```rb
run(sequence: {
```

File: library/sequencer/src/lib/ui/sequence.rb [hunk 4]
```diff
@@ -24,13 +91,38 @@ def self.run(aliases, sequence)
...
+    #     START => "step1",
+    #     "step1" => { next: "step2" },
+    #     "step2" => { next: :finish }
+    #   }
+    #   output = {
+    #     START => "step1",
+    #     "step1" => { next: "step2", abort: :abort },
+    #     "step2" => { next: :finish, abort: :abort }
+    #   }
+    #   input == output # => true
```
Comment (mvidner) [inline library/sequencer/src/lib/ui/sequence.rb]:
```rb
abortable(input) == output
```

File: library/sequencer/src/lib/ui/sequence.rb (hunk 1 above)
Comment (jreidinger) [reply library/sequencer/src/lib/ui/sequence.rb]: well, when I thinking about this it does not need to be UI. In fact it is just some kind of StateMachine with defined transitions between states.

**Pull request merged**